handbook/engineering/onboarding.md

```diff
@@ -15,29 +40,62 @@ Welcome to Sourcegraph! This document will guide you through engineering specifi
   - [Site24x7](https://www.site24x7.com) (optional; recommended for Distribution team members)
   - [HoneyComb.io](https://www.honeycomb.io/)
   - Ask Christina to send an invite to [Productboard](https://sourcegraph.productboard.com)
-- Schedule a recurring [1-1](../leadership/1-1.md).
-- Schedule time to discuss Sourcegraph's system architecture and tech stack.
-- Schedule a time to discuss on-call rotation.
-- Assign one or more starter tasks (e.g. small bugs or issues).
 
-## Engineer checklist
+## Weeks 1-3
 
-As you are completing these tasks, you may notice documentation, processes, or code that is broken or out of date. Your first priority is to fix these things. **Act like you own the onboarding experience of the next engineer that we hire.**
+### Starter tasks
 
-- [Complete general onboarding](../people-ops/onboarding.md#for-all-new-teammates)
-- [Configure your GitHub notifications.](github-notifications/index.md)
-- Message each person on your immediate team to setup a time to meet them. Some things you might want to learn:
-  - What brought them to Sourcegraph?
-  - What are they currently working on?
-  - What do they do for fun?
-- Join #dev-announce, #dev-chat, and your team's channel on Slack. [Team chat documentation](../communication/team_chat.md#engineering)
+Your manager will assign to you three starter tasks that you should aim to complete in your first three weeks. These tasks are small, atomic, and intended to expose you to different parts of our codebase and product: it's important that you build the flexibility to fix any problem you'll be faced with at Sourcegraph, and don't narrow down your comfort zone to a certain part of our codebase or product.
+
+As you're working on these tasks:
```

Comment:
When I joined @beyang said to me: "In your first few weeks: optimise for learning" @nicksnyder said something along the same lines and it really stuck with me. It takes a lot of the "i need to ship asap" pressure of when starting.

Maybe we can add something like this to this section?

Reply:
What do you think of this?

```suggestion
As you're working on these tasks:
- Optimize for learning: your priority is to soak up as much context as possible.
```

Reply:
Yep, that's good, or go even further:

```suggestion
As you're working on these tasks:
- Optimize for learning: your priority is to soak up as much context as possible, not to ship something as fast as possible.
```

I'll let you decide, though.